stackit/internal/services/iaas/securitygroup/resource.go

```diff
@@ -0,0 +1,446 @@
+			config.WithRegion(providerData.Region),
+		)
+	}
+
+	if err != nil {
+		core.LogAndAddError(ctx, &resp.Diagnostics, "Error configuring API client", fmt.Sprintf("Configuring client: %v. This is an error related to the provider configuration, not to the resource configuration", err))
+		return
+	}
+
+	r.client = apiClient
+	tflog.Info(ctx, "iaasalpha client configured")
+}
+
+// Schema defines the schema for the resource.
+func (r *securityGroupResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: features.AddBetaDescription("Security group resource schema. Must have a `region` specified in the provider configuration."),
+		Description:         "Security group resource schema. Must have a `region` specified in the provider configuration.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description: "Terraform's internal resource ID. It is structured as \"`project_id`,`security_group_id`\".",
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"project_id": schema.StringAttribute{
+				Description: "STACKIT project ID to which the security group is associated.",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.String{
+					validate.UUID(),
+					validate.NoSeparator(),
+				},
+			},
+			"security_group_id": schema.StringAttribute{
+				Description: "The security group ID.",
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+				Validators: []validator.String{
+					validate.UUID(),
+					validate.NoSeparator(),
+				},
+			},
+			"name": schema.StringAttribute{
+				Description: "The name of the security group.",
+				Required:    true,
+				Validators: []validator.String{
+					stringvalidator.LengthAtLeast(1),
+					stringvalidator.LengthAtMost(63),
+					stringvalidator.RegexMatches(
+						regexp.MustCompile(`^[A-Za-z0-9]+((-|_|\s|\.)[A-Za-z0-9]+)*$`),
+						"must match expression"),
+				},
+			},
+			"description": schema.StringAttribute{
+				Description: "The description of the security group.",
+				Optional:    true,
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+				Validators: []validator.String{
+					stringvalidator.LengthAtLeast(1),
+					stringvalidator.LengthAtMost(127),
+				},
+			},
+			"labels": schema.MapAttribute{
+				Description: "Labels are key-value string pairs which can be attached to a resource container",
+				ElementType: types.StringType,
+				Optional:    true,
+			},
+			"stateful": schema.BoolAttribute{
+				Description: "Shows if a security group is stateful or stateless. There can only be one security group per network interface/server.",
```

Comment:
```suggestion
				Description: "Configures if a security group is stateful or stateless.",
```

About "There can only be one security group per network interface/server.", I don't fully get it, can you clarify with the team what it means?

Is that sentence only relevant for stateful security groups?

Reply:
Description has been changed in the meantime in API docs. It is now: "Shows if a security group is stateful or stateless. You can only have one type of security groups per network interface/server."

The limitation is not one security group per network interface. It is one type of security group. So in a network interface, there can be either stateful or stateless security groups. 

Would that be more understandable if I changed to : 

"Configures if a security group is stateful or stateless. There can only be one type of security groups per network interface/server."

Reply:
Yes, that looks good